shared/src/main/scala/mlscript/ucs/stages/Transformation.scala

```diff
@@ -105,26 +105,27 @@ trait Transformation { self: mlscript.pretyper.Traceable =>
     }
   }
 
-  private def rec(term: Term)(implicit useNewDefs: Bool): Pattern = term match {
+  private def transformPattern(term: Term)(implicit useNewDefs: Bool): Pattern = term match {
+    case nme @ Var("true" | "false") => ConcretePattern(nme)
     case nme @ Var(name) if name.headOption.exists(_.isUpper) => ClassPattern(nme, N)
     case nme: Var => NamePattern(nme)
     case literal: Lit => LiteralPattern(literal)
     case App(classNme @ Var(_), Tup(parameters)) =>
       ClassPattern(classNme, S(parameters.map {
         case (_, Fld(_, Var("_"))) => N // Consider "_" as wildcard.
-        case (_, Fld(_, t)) => S(rec(t))
+        case (_, Fld(_, t)) => S(transformPattern(t))
       }))
     case Tup(fields) => TuplePattern(fields.map {
       case _ -> Fld(_, Var("_")) => N // Consider "_" as wildcard.
-      case _ -> Fld(_, t       ) => S(rec(t))
+      case _ -> Fld(_, t       ) => S(transformPattern(t))
```

Comment:
Maybe document this part of the code:
```suggestion
      case _ -> Fld(_, t       ) => S(transformPattern(t))
      //                ^^^^^^^  space reserved to park my car
```